File: app/src/main/java/com/dm/bomber/services/core/Phone.java
package com.dm.bomber.services.core;

import androidx.annotation.NonNull;

public class Phone {

    private final String countryCode;
    private final String phone;

    public Phone(String countryCode, String phone) {
        this.countryCode = countryCode;
        this.phone = phone;
    }

    public static String format(String phone, String mask) {
        StringBuilder formattedPhone = new StringBuilder();

        int index = 0;
        for (char symbol : mask.toCharArray())
            if (index < phone.length())
                formattedPhone.append(symbol == '*' ? phone.charAt(index++) : symbol);

        return formattedPhone.toString();
    }

    public String format(String mask) {
        return format(phone, mask);
    }

    public String getPhone() {
        return phone;
    }

    public String getCountryCode() {
        return countryCode;
    }

    @NonNull
    public String toString() {
        return countryCode + phone;
    }
}

File: app/src/main/java/com/dm/bomber/services/core/Phone.kt
package com.dm.bomber.services.core

class Phone(
    val countryCode: String,
    val phone: String
) {
    fun format(mask: String): String {
        return format(phone, mask)
    }

    override fun toString(): String {
        return countryCode + phone
    }

    companion object {
        fun format(phone: String, mask: String): String {
            val formattedPhone = StringBuilder()
            var index = 0

            for (symbol in mask.toCharArray())
                if (index < phone.length)
                    formattedPhone.append(if (symbol == '*') phone[index++] else symbol)

            return formattedPhone.toString()
        }
    }
}
